src/EPR.SubsidiaryBulkUpload.Application/Services/NotificationService.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Text.Json;
+using EPR.SubsidiaryBulkUpload.Application.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using StackExchange.Redis;
+
+namespace EPR.SubsidiaryBulkUpload.Application.Services;
+
+public class NotificationService(
+    ILogger<NotificationService> logger, IConnectionMultiplexer redisConnectionMultiplexer) : INotificationService
+{
+    private readonly ILogger<NotificationService> _logger = logger;
+    private readonly IDatabase _redisDatabase = redisConnectionMultiplexer.GetDatabase();
+
+    public async Task SetStatus(string key, string status)
+    {
+        await _redisDatabase.StringSetAsync(key, status);
+
+        _logger.LogInformation("Redis updated key: {key} status: {status}", key, status);
+    }
+
+    public async Task SetErrorStatus<T>(string key, List<T> errorsModel)
```

Comment:
I see that this is templated (which I like by the way), but should it be a well known model?
How will the front end know how to display the errors?

Reply:
The front end will get the errors from the web gateway api, so if we have a defined model here it will be easy to copy the same model class in there and the front end.

Reply:
Updated.